src/Modules/OrchardCore.Commerce/Drivers/OrderPartDisplayDriver.cs

```diff
@@ -106,14 +106,14 @@ public override async Task<IDisplayResult> UpdateAsync(OrderPart part, IUpdateMo
         var orderLineItems = new List<OrderLineItem>();
         foreach (var lineItem in viewModelLineItems)
         {
-            var lineItemProductSku = lineItem.ProductSku.ToUpperInvariant();
+            var lineItemProductSku = lineItem.ProductSku?.ToUpperInvariant();
 
             // If the provided SKU does not belong to an existing Product content item, it should not be added.
-            if (await _productService.GetProductAsync(lineItemProductSku) is not { } productPart)
+            if (string.IsNullOrEmpty(lineItemProductSku) || await _productService.GetProductAsync(lineItemProductSku) is not { } productPart)
             {
                 updater.ModelState.AddModelError(
                     modelErrorKey,
-                    T["SKU {0} does not belong to an existing Product.", lineItemProductSku]);
+                    T["SKU \"{0}\" is empty or does not belong to an existing Product.", lineItemProductSku]);
 
                 continue;
             }
```

Comment:
I think it would be nicer to split these two conditions into two separate error messages.
The `SKU "" is empty or does not belong to an existing Product.` looks very clunky to me.